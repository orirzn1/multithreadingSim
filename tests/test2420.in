L15
S5
O6
N7

T0
I@0x00000000
LOAD $2, $4, $4
SUBI $4, $6, 0x4
ADDI $0, $4, 0x2
ADD $0, $1, $4
HALT $0

T1
I@0x00000000
SUB $4, $7, $1
SUB $2, $4, $0
SUB $5, $7, $7
ADDI $6, $4, 0x3
SUB $7, $4, $1
SUBI $1, $4, 0x4
SUBI $1, $6, 0x4
SUBI $0, $0, 0x3
HALT $0

T2
I@0x00000000
SUBI $0, $4, 0x1
SUB $4, $3, $0
STORE $2, $2, 0x3
LOAD $0, $3, 0x1
STORE $3, $4, 0x1
SUBI $5, $6, 0x2
HALT $0

T3
I@0x00000000
ADDI $6, $5, 0x4
SUBI $0, $2, 0x1
STORE $1, $7, $7
SUBI $3, $1, 0x3
SUB $2, $1, $2
SUBI $0, $4, 0x2
SUB $3, $6, $6
ADDI $3, $0, 0x1
SUBI $1, $3, 0x1
HALT $0

T4
I@0x00000000
ADD $3, $0, $4
ADD $1, $0, $1
SUBI $3, $5, 0x1
STORE $7, $7, $0
SUBI $5, $4, 0x3
ADD $6, $5, $5
ADDI $4, $6, 0x2
ADDI $7, $0, 0x4
HALT $0

T5
I@0x00000000
SUBI $6, $0, 0x3
SUB $2, $1, $1
STORE $0, $1, 0x2
ADD $3, $1, $5
HALT $0

T6
I@0x00000000
ADD $1, $0, $1
ADDI $6, $5, 0x2
SUBI $1, $1, 0x3
SUBI $0, $1, 0x4
ADD $2, $3, $2
LOAD $2, $7, 0x1
ADDI $2, $3, 0x3
SUBI $1, $4, 0x1
ADD $7, $0, $2
HALT $0

D@0x0
0x7
0x3
0x9
0x9
0x4
0x8
0xA
0x7
0x2
