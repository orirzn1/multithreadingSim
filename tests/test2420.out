
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x7	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFF	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0xFFFFFFFE	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFC	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFA	R7 = 0x4
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Register file thread id 6:
	R0 = 0xFFFFFFF9	R1 = 0xFFFFFFFF	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0xFFFFFFFC
Blocked MT CPI for this program 2.509091

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0xFFFFFFFC	R1 = 0x0	R2 = 0x1	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFF	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x1	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0xFFFFFFFE	R1 = 0xFFFFFFFE	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0xFFFFFFFC	R5 = 0xFFFFFFFD	R6 = 0xFFFFFFFA	R7 = 0x4
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFD	R7 = 0x0
Register file thread id 6:
	R0 = 0xFFFFFFF9	R1 = 0xFFFFFFFF	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x2	R7 = 0xFFFFFFFC
Finegrained Multithreading CPI for this program 1.018182

